mongo.go

```diff
@@ -57,8 +57,16 @@ func (mongo mongoConfig) dump(sb s3Backend, keyPath, database string) error {
 
 	log.Debugf("Mongo archive file %v ready for writing", mongoArchive)
 
-	key := getKey(keyPath)
-	e, err := newEncryptor(key, wr)
+	privateKey, publicKeyList, err := getKeys(publicKeyPath)
+	if err != nil {
+		log.Error("Could not retrieve public key or generate private key")
+
+		return err
+	}
```

Comment:
```suggestion
	if err != nil {
		return err
	}
```
No point of logging an error here since `err` will be propagated to main and logged as `fatal`

Reply:
same